src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+      .attribute(MaxLikelihoodChangeFormatTag, info.delta)
+      .attribute(MecReductionFormatTag, info.rMEC)
+  }
+}
+
+/** Information about a variant phased by HapCut.
+  *
+  * @param offset the 1-based offset of the variant relative to the variants given to HapCut.
+  * @param haplotype1 the first allele.
+  * @param haplotype2 the second allele.
+  * @param contig the contig.
+  * @param pos the position.
+  * @param ref the reference allele.
+  * @param alts the alternate alleles.
+  * @param genotype the genotype.
+  * @param info the HapCut-specific genotype information.
+  * @param firstInBlock true if this variant is the first variant in a phased block, false otherwise.
+  */
+private class HapCutCall private(val offset: Int,
+                                 val haplotype1: Int, val haplotype2: Int,
+                                 val contig: String, val pos: Int,
+                                 val ref: String, val alts: String,
+                                 val genotype: String, val info: HapCutCallGenotypeInfo,
+                                 val firstInBlock: Boolean
+                                )
+
+private object HapCutCall {
+
+  /** Parse a variant line. */
+  def apply(callLine: String, blockOffset: Int): HapCutCall = {
+    val tokens = callLine.split("\t")
+    val offset = tokens(0).toInt
+    new HapCutCall(
+      offset       = offset,
+      haplotype1   = tokens(1).toInt,
+      haplotype2   = tokens(2).toInt,
+      contig       = tokens(3),
+      pos          = tokens(4).toInt,
+      ref          = tokens(5),
+      alts         = tokens(6),
+      genotype     = tokens(7),
+      info         = HapCutCallGenotypeInfo(tokens(8)),
+      firstInBlock = blockOffset == offset
+    )
+  }
+
+  /** Converts the HapCut variant representation to a VariantContext. */
+  def toVariantContext(call: HapCutCall, sampleName: String): VariantContext = {
+    // Parse the alleles
+    val refAllele = Allele.create(call.ref, true)
+    val altAlleles = call.alts.split(",").map { alt =>
+      Allele.create(alt, false)
+    }.toSeq
+    val alleles = refAllele +: altAlleles
+    val allelesCollection = alleles.asJavaCollection
+
+    // Assumes the genotype is before the first ":" token
+    val genotypeAlleles = call.genotype.split(":").head.split("/").map(_.toInt).map(i => alleles(i))
+    val genotype = HapCutCallGenotypeInfo.addTo(new GenotypeBuilder(sampleName, genotypeAlleles.toList), call.info).make()
+
+    // build the context, and make sure to recompute the end
+    new VariantContextBuilder(
+      s"${call.offset}",
+      call.contig,
+      call.pos.toLong,
+      call.pos.toLong,
+      allelesCollection
+    )
+      .computeEndFromAlleles(seqAsJavaList(alleles.toSeq), call.pos)
+      .genotypes(genotype).make()
+  }
+}
+
+/** Reads phased blocks and variants from a HapCut output file, returning a HapCut call for each input variant call to
```

Comment:
Something here (and in the usage of HapCutReader) doesn't make sense to me.  The comment says it returns an HapCut call for each input variant, then turns around and says that if HapCut didn't phase a variant `None` is returned because it's not present in the HapCut output.  But this reader class doesn't take in anything _but_ the HapCut output, so how can it know about variants that were not present in the output?